compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt

```diff
@@ -13,12 +13,112 @@ import org.jetbrains.kotlin.psi.psiUtil.getValueParameters
 import org.jetbrains.kotlin.psi.psiUtil.modalityModifierType
 import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierType
 
+/**
+ * [classOrObject] is a function that intercepts all [KtClass] elements that [match]
+ * allowing a [Transform] to change the intercepted AST tree before compilation.
+ *
+ * An extension function of [Meta] and inheriting from [ExtensionPhase], [classOrObject] was designed to to feed in
+ * virtually any kind of [KtClass] predicate, followed by mapping a function that takes the desired [Scope] of our
+ * [KtClass] to change whatever PSI elements desired.
+ *
+ * For example, the [LensPlugin] and the [HigherKindPlugin] favor easy update of immutable data structures and ad-hoc polymorphism that does not require inheritance.
+ *
+ * In the [LensPlugin], we check to see if a type is a product type.  To add context, the cardinality of a product
+ * type is the product of its contained properties:
+ *
+ * ```
+ *      |(a, b)| = |a| x |b|
+ * ```
+ *
+ * In particular, the [LensPlugin] is passing a function to whatever class or object type is a product type:
+ *
+ * ```
+ *  val Meta.lenses: Plugin
+ *    get() =
+ *      "lenses" {
+ *        meta {
+ *          classOrObject(::isProductOrType) { c -> ... }
+ *        }
+ *      }
+ * ```
+ *
+ * Where the function type for [::isProductType] is written as:
+ *
+ * ```
+ * fun isProductType(ktClass: KtClass): Boolean =
+ *      ktClass.isData() &&
+ *        ktClass.primaryConstructorParameters.isNotEmpty() &&
+ *        ktClass.primaryConstructorParameters.all { !it.isMutable } &&
+ *        ktClass.typeParameters.isEmpty()
+ * ```
+ *
+ * For a [KtClass] to be a product type, we check to see if the [KtClass] is an ADT, and that the cardinality of
+ * the constructor parameters is greater than one and non-mutable. For the [LensPlugin], we've just demonstrated how we
+ * can check for the product type of a [KtClass] by accessing the properties made available by the Kotlin PSI tree.
+ *
+ * In the example provided, we favor and talk about writing compiler plugins for functional stuff,
+ * which is great - I wanted to emphasize that folks really don't need to worry about writing predicates
```

Comment:
```suggestion
 * which is great - We'd want to emphasize that folks really don't need to worry about writing predicates
```
nit: To me it's nice if we talk about "us" as a team in the docs, so there's an overall language coherence and they don't look as if they were written by tons of different devs, even if that's true.

Reply:
Thank you for that catch!!